chromium_src/components/sessions/content/content_serialized_navigation_driver.cc

```diff
@@ -0,0 +1,25 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "components/sessions/content/content_serialized_navigation_driver.h"
+#include "components/sessions/core/serialized_navigation_entry.h"
+#include "content/public/common/url_constants.h"
+
+#define GetSanitizedPageStateForPickle \
+  GetSanitizedPageStateForPickle_ChromiumImpl
+#include "src/components/sessions/content/content_serialized_navigation_driver.cc"
+#undef GetSanitizedPageStateForPickle
+
+namespace sessions {
+std::string ContentSerializedNavigationDriver::GetSanitizedPageStateForPickle(
+    const sessions::SerializedNavigationEntry* navigation) const {
+  if (navigation->virtual_url().SchemeIs(content::kChromeUIScheme))
+    return std::string();
```

Comment:
this seems rather heavy-handed, shouldn't we do something like `PageState::RemovePasswordData`?

Reply:
I'm worried about what else this might negatively impact

Reply:
`PageState::RemovePasswordData()` should already be called in `GetSanitizedPageStateForPickle`; maybe it would make sense to override that instead?
https://source.chromium.org/chromium/chromium/src/+/main:components/sessions/content/content_serialized_navigation_driver.cc;l=57-66;drc=5d77f7207bf888df5456310dff690bb183b60a73

```cpp
std::string
ContentSerializedNavigationDriver::GetSanitizedPageStateForPickle(
    const SerializedNavigationEntry* navigation) const {
  if (!navigation->has_post_data())
    return navigation->encoded_page_state();

  blink::PageState page_state =
      blink::PageState::CreateFromEncodedData(navigation->encoded_page_state());
  return page_state.RemovePasswordData().ToEncodedData();
}
```

Reply:
Actually, reading the issue:
https://github.com/brave/brave-browser/issues/24625

I think this solution will work well. With this, we'll move to not saving page state for Chrome internal pages. I would be interested to know if there was a page we don't want to do this on? I tried the settings page and doing a search; it still restores the search keyword also

Reply:
search keyword is restored from url query part

Reply:
This change made regression - https://github.com/brave/brave-browser/issues/30890#issuecomment-1586645164

Reply:
This change (or I guess the fix above) seems to have indirectly led to https://github.com/brave/brave-browser/issues/31237, which is a fairly common scenario in Rewards. We may want to rethink the overall approach here (or at the very least tighten up the logic).